hack/ensure-go.sh

```diff
@@ -31,7 +31,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(go version)"
   local minimum_go_version
-  minimum_go_version=go1.17.0
+  minimum_go_version=go1.18.4
```

Comment:
CAPI has this value as `go1.18.3`. While it's wise to be on the latest patch, I think we should match CAPI's minimum version so we don't impose additional developer requirements. If you're successfully writing code for CAPI with Go 1.18.3, that should also work for CAPZ in my opinion.

Reply:
Fixed.